tests/cases/fourslash/completionsSelfDeclaring2.ts

```diff
@@ -0,0 +1,18 @@
+/// <reference path="fourslash.ts" />
+
+////function f1<T>(x: T) {}
+////f1({ abc/*1*/ });
```

Comment:
This change looks great but do you think we should allow `abc` in completions if `abc` is some variable in scope?

Reply:
Right, I guess don't filter our anything that could be a shorthand property declaration.

Reply:
Those were never actually provided as object literal member completions, so this PR doesn’t filter them per se. I’m not sure if anything is really needed here as those completions already come up as plain text matches (I think this is provided by Monaco / VS):

<img width="502" alt="image" src="https://user-images.githubusercontent.com/3277153/71100023-21c40a80-2169-11ea-886a-f05b7484f7d0.png">

I’m not sure if it’s right to make all in-scope identifiers first-class completions. _Real_ missing members from the contextual type are a very strong signal that the user will want to add those members, but the presence of random identifiers is a very weak signal. We can sort them appropriately, but otherwise there’s no visual distinction that one completion is a sure thing and the other is a wild guess. I think this would get really confusing.

Besides, as I learned in #34855, with generics it can be extremely difficult to know ahead of time whether some hypothetical arbitrary member is a valid completion or not. It often appears not to be until the user adds it, when inference runs and incorporates it into the contextual type just because it’s there. But with conditional types, we effectively can’t know whether that member is going to be valid without speculatively type checking the change, which is prohibitively expensive.